sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -958,6 +958,18 @@ case class Sort(
   override protected def withNewChildInternal(newChild: LogicalPlan): Sort = copy(child = newChild)
 }
 
+/**
+ * Clustering data within the partition.
+ *
+ * @param cluster The clustering expressions
+ * @param child   Child logical plan
+ */
+case class Clustering(cluster: Seq[SortOrder], child: LogicalPlan) extends UnaryNode {
```

Comment:
shall we split it into `clusteringKeys: Seq[Expression], sortKeys: Seq[SortOrder]`?